sw/device/tests/aes_masking_off_test.c

```diff
@@ -0,0 +1,140 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "hw/ip/aes/model/aes_modes.h"
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/base/multibits.h"
+#include "sw/device/lib/dif/dif_aes.h"
+#include "sw/device/lib/dif/dif_csrng.h"
+#include "sw/device/lib/dif/dif_csrng_shared.h"
+#include "sw/device/lib/dif/dif_edn.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/aes_testutils.h"
+#include "sw/device/lib/testing/csrng_testutils.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+#define TIMEOUT (1000 * 1000)
+
+// The mask share, used to mask kKey. Note that the masking should not be done
+// manually. Software is expected to get the key in two shares right from the
+// beginning.
+static const uint8_t kKeyShare1[] = {
+    0x0f, 0x1f, 0x2f, 0x3F, 0x4f, 0x5f, 0x6f, 0x7f, 0x8f, 0x9f, 0xaf,
+    0xbf, 0xcf, 0xdf, 0xef, 0xff, 0x0a, 0x1a, 0x2a, 0x3a, 0x4a, 0x5a,
+    0x6a, 0x7a, 0x8a, 0x9a, 0xaa, 0xba, 0xca, 0xda, 0xea, 0xfa,
+};
+
+OTTF_DEFINE_TEST_CONFIG();
+
+bool test_main(void) {
+  /*
+   * Test seed generation using CSRNG SW application interface.
+   *
+   * We try to generate the seed leading to an all-zero output of the AES
+   * masking PRNG using the CSRNG SW application interface. Unlike the HW
+   * application interfaces of CSRNG that connect to the EDNs, this interface
+   * allows for probing the internal state of the CSRNG instance after
+   * individual commands. This gives us higher visibility. If for some reason
+   * this phase isn't succesful, we don't even need to try it on the HW
+   * interfaces.
+   */
+  LOG_INFO("Testing CSRNG SW application interface");
+
+  // Disable EDN connected to AES as well as CSRNG.
+  const dif_edn_t edn = {
+      .base_addr = mmio_region_from_addr(TOP_EARLGREY_EDN0_BASE_ADDR)};
+  const dif_csrng_t csrng = {
+      .base_addr = mmio_region_from_addr(TOP_EARLGREY_CSRNG_BASE_ADDR)};
+  CHECK_DIF_OK(dif_edn_stop(&edn));
+  CHECK_DIF_OK(dif_csrng_stop(&csrng));
+
+  // Re-eanble CSRNG.
+  CHECK_DIF_OK(dif_csrng_configure(&csrng));
+
+  // Perform seed generation as well as checking of internal state and outputs.
+  aes_testutils_debug_csrng_instantiate(&csrng);
+  aes_testutils_debug_csrng_generate(&csrng);
+  aes_testutils_debug_csrng_reseed(&csrng);
+  aes_testutils_debug_csrng_generate(&csrng);
+
+  /*
```

Comment:
To follow the style guide, in-code comments should use the `// Comment.` format.

Reply:
Done, thanks for pointing that out, I wasn't aware of this.